Python tests fail in CI on macOS

<!-- If you're not sure whether your problem is a bug, please ask a question at
http://answers.gazebosim.org instead.-->

## Environment
* OS Version: macOS
* Source or binary build? binary builds of gz-sim7, gz-sim8


## Description
* Expected behavior: python tests should pass
* Actual behavior: python tests fail

gz-sim7: from https://build.osrfoundation.org/job/gz_sim-ci-gz-sim7-homebrew-amd64/4/

~~~
test 301
        Start 301: testFixture_TEST

301: Test command: /usr/local/Frameworks/Python.framework/Versions/3.12/bin/python3.12 "/Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim7-homebrew-amd64/gz-sim/python/test/testFixture_TEST.py"
301: Working Directory: /Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim7-homebrew-amd64/build/python
301: Environment variables: 
301:  PYTHONPATH=/usr/local/Cellar/gz-sim7/HEAD/lib/python/
301:  LD_LIBRARY_PATH=/usr/local/Cellar/gz-sim7/HEAD/lib:
301: Test timeout computed to be: 1500
301: Traceback (most recent call last):
301:   File "/Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim7-homebrew-amd64/gz-sim/python/test/testFixture_TEST.py", line 21, in <module>
301:     from gz.math7 import Vector3d
301: ModuleNotFoundError: No module named 'gz.math7'
301/301 Test #301: testFixture_TEST ........................................***Failed    1.07 sec
~~~

gz-sim8: from https://build.osrfoundation.org/job/gz_sim-ci-gz-sim8-homebrew-amd64/4/

~~~
...
        Start 320: world_TEST

320: Test command: /usr/local/Frameworks/Python.framework/Versions/3.12/bin/python3.12 "/Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim8-homebrew-amd64/gz-sim/python/test/world_TEST.py"
320: Working Directory: /Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim8-homebrew-amd64/build/python
320: Environment variables: 
320:  PYTHONPATH=/usr/local/Cellar/gz-sim8/HEAD/lib/python/
320:  LD_LIBRARY_PATH=/usr/local/Cellar/gz-sim8/HEAD/lib:
320: Test timeout computed to be: 1500
320: Traceback (most recent call last):
320:   File "/Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim8-homebrew-amd64/gz-sim/python/test/world_TEST.py", line 21, in <module>
320:     from gz_test_deps.math import SphericalCoordinates, Vector3d
320:   File "/Users/jenkins/jenkins-agent/workspace/gz_sim-ci-gz-sim8-homebrew-amd64/gz-sim/python/test/gz_test_deps/math.py", line 1, in <module>
320:     from gz.math7 import *
320: ModuleNotFoundError: No module named 'gz.math7'
320/320 Test #320: world_TEST ..............................................***Failed    0.76 sec
...
The following tests FAILED:
	305 - INTEGRATION_python_system_loader (Failed)
	313 - actor_TEST (Failed)
	314 - joint_TEST (Failed)
	315 - light_TEST (Failed)
	316 - link_TEST (Failed)
	318 - sensor_TEST (Failed)
	319 - testFixture_TEST (Failed)
	320 - world_TEST (Failed)
~~~

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. Run a macOS CI build
2.
3.

## Output
<!-- Provide screenshots, console logs, backtraces, and/or anything that could
be useful to us resolving this issue -->

Hey! I noticed the following log output in [gz-sim-main](https://build.osrfoundation.org/job/gz_sim-ci-main-homebrew-amd64/21/consoleFull#-394183657aa60f765-a60a-427d-900c-dc128ab22287) and [gz-launch-main](https://build.osrfoundation.org/job/gz_launch-ci-main-homebrew-amd64/16/consoleFull#-394183657aa60f765-a60a-427d-900c-dc128ab22287):

```
CMake Error at /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Python3 (missing: Development Development.Module
  Development.Embed) (found version "3.12")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPython/Support.cmake:3867 (find_package_handle_standard_args)
  /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPython3.cmake:545 (include)
  CMakeLists.txt:151 (find_package)
```

I think this is related to 3.12 -> 3.11 changes, right?

> Hey! I noticed the following log output in [gz-sim-main](https://build.osrfoundation.org/job/gz_sim-ci-main-homebrew-amd64/21/consoleFull#-394183657aa60f765-a60a-427d-900c-dc128ab22287) and [gz-launch-main](https://build.osrfoundation.org/job/gz_launch-ci-main-homebrew-amd64/16/consoleFull#-394183657aa60f765-a60a-427d-900c-dc128ab22287):
> 
> ```
> CMake Error at /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
>   Could NOT find Python3 (missing: Development Development.Module
>   Development.Embed) (found version "3.12")
> Call Stack (most recent call first):
>   /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
>   /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPython/Support.cmake:3867 (find_package_handle_standard_args)
>   /usr/local/Cellar/cmake/3.28.2/share/cmake/Modules/FindPython3.cmake:545 (include)
>   CMakeLists.txt:151 (find_package)
> ```
> 
> I think this is related to 3.12 -> 3.11 changes, right?

I think it will be fixed by https://github.com/gazebo-tooling/release-tools/pull/1110, but I'm still waiting for an approval on that PR

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -244,7 +244,7 @@ add_subdirectory(examples)
 #============================================================================
 gz_create_packages()
 
-if (pybind11_FOUND)
+if (pybind11_FOUND AND NOT SKIP_PYBIND11)
 	add_subdirectory(python)
 endif()
 #============================================================================

